### PR TITLE
factor out mutex pkg

### PR DESCRIPTION
**Pull request by kshen0**
## Overview
Factors out `Mutex` from `libs/custom-scanner` for use in other packages.

## Demo Video or Screenshot
N/A

## Testing Plan
Moved mutex tests to mutex package. No new tests added.

## Checklist

- [x] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

### CODE REVIEW

File: libs/mutex/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ */
+module.exports = {
+  ...shared,
+  collectCoverageFrom: [
+    ...shared.collectCoverageFrom,
+    '!src/**/index.ts',
+    '!src/cli/**/*.ts',
+    '!src/mocks/**/*.ts',
+    '!src/types/**/*.ts',
+  ],
```
Comment (adghayes) [inline libs/mutex/jest.config.js:14]: could remove / simplify this

File: libs/mutex/package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "@votingworks/mutex",
+  "version": "1.0.0",
+  "description": "Generic mutex package for controlling access to a shared resource.",
+  "keywords": [],
+  "license": "AGPL-3.0",
+  "author": "",
+  "main": "build/index.js",
+  "scripts": {
+    "analyze-packets": "cd tools/analyze-packets && vite",
```
Comment (adghayes) [inline libs/mutex/package.json:10]: could probably remove this script

**Review by adghayes — COMMENTED** [review]
Is there a compelling reason that this needs to be its own package rather than exist in `basics` or `utils`? With what I know, I think this would be better there.

Because of the way our CircleCI works, this will spin up an entire other instance on every run just to run the couple tests in `mutex.test`. And it makes our dependency tree just a little more complicated.

**Comment by kshen0** [comment]
Fair point, I'll move it to utils and that should also address the config bloat you commented on.

**Review by adghayes — APPROVED** [review]
Thanks for making the adjustment!

File: libs/custom-scanner/src/custom_a4_scanner.ts [hunk 3]
```diff
@@ -569,7 +568,7 @@ export class CustomA4Scanner implements CustomScanner {
   /**
    * Resets the hardware.
    */
-  async resetHardware(): Promise<Result<void, ErrorCode>> {
+  resetHardware(): Promise<Result<void, ErrorCode>> {
```
Comment (kshen0) [inline libs/custom-scanner/src/custom_a4_scanner.ts:571]: I'm not sure why these functions were declared as `async` when they don't have any `await` expressions. Our linter prohibits this so I went ahead and removed the `async` declarations.

**Pull request merged**